include/swift/AST/DiagnosticsSema.def

```diff
@@ -4523,6 +4523,12 @@ ERROR(override_nsobject_hashvalue_error,none,
       "'NSObject.hashValue' is not overridable; "
       "did you mean to override 'NSObject.hash'?", ())
 
+WARNING(hashable_customEquatable, none,
+        "Automatically generated 'Hashable' implementation for type %0 "
+        "may not match the behavior of custom '==' operator",
+        (Type))
```

Comment:
1. Can you give the diagnostic identifier a more descriptive name? Also, please use lowercase words only. For example: `synthesized_hashable_may_not_match_custom_equatable`.
2. Diagnostic text must start with a lowercase letter.
3. Can you move the argument (aka `(Type)`) to the line above since it can fit there?